pkg/types/auth_webhook.go

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2021 The Yorkie Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package types
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+)
+
+// VerbType represents an action taken on the document.
+type VerbType string
+
+const (
+
+	// Read represents the case of only reading the given document.
+	Read VerbType = "r"
+
+	// ReadWrite represents the case of reading and writing the given document.
+	ReadWrite VerbType = "rw"
+)
+
+var (
+	// ErrInvalidWebhookRequest is returned when the given webhook request is not valid.
+	ErrInvalidWebhookRequest = errors.New("invalid authorization webhook request")
+
+	// ErrInvalidWebhookResponse is returned when the given webhook response is not valid.
+	ErrInvalidWebhookResponse = errors.New("invalid authorization webhook response")
+)
+
+// Method represents a method name of RPC.
+type Method string
+
+// Belows are the names of RPCs.
+const (
+	ActivateClient   Method = "ActivateClient"
+	DeactivateClient Method = "DeactivateClient"
+	AttachDocument   Method = "AttachDocument"
+	DetachDocument   Method = "DetachDocument"
+	PushPull         Method = "PushPull"
+)
+
+// AccessAttribute represents an access attribute.
+type AccessAttribute struct {
+	Key  string   `json:"key"`
+	Verb VerbType `json:"verb"`
+}
+
+// AccessInfo represents an access information.
+type AccessInfo struct {
+	Method     Method
+	Attributes []AccessAttribute
+}
+
+// AuthWebhookRequest represents the request of authentication webhook.
+type AuthWebhookRequest struct {
+	Token      string            `json:"token"`
+	Method     Method            `json:"method"`
+	Attributes []AccessAttribute `json:"attributes"`
+}
+
+// NewAuthWebhookRequest creates a new instance of AuthWebhookRequest.
+func NewAuthWebhookRequest(reader io.Reader) (*AuthWebhookRequest, error) {
+	req := &AuthWebhookRequest{}
+
+	bytes, err := ioutil.ReadAll(reader)
```

Comment:
How about using `io.Copy`??🧐
I think this part is likely to be called frequently
https://daryl-ng.medium.com/why-you-should-avoid-ioutil-readall-in-go-e6be4de180f8

Reply:
@ppeeou I think your opinion is meaningful.
So I changed `ioutil.ReadAll` to `Decoder.Decode` by referring to the link below.
https://stackoverflow.com/questions/52539695/alternative-to-ioutil-readall-in-go/52540512